### PR TITLE
Flattening app-operator

### PR DESCRIPTION
**Pull request by tomahawk28**
Toward https://github.com/giantswarm/giantswarm/issues/7895

Deleting deployment by draughtsman part and uploading charts on app collections.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -67,10 +57,57 @@ workflows:
...
+              ignore: /.*/
+            tags:
+              only: /^v.*/
+
+      # deploy to azure installations (only tags)
+      - architect/push-to-app-collection:
+          name: push-app-operator-to-azure-app-collection
+          app_name: "app-operator"
+          app_collection_repo: "azure-app-collection"
+          unique: "true"
```
Comment (tomahawk28) [inline .circleci/config.yml:92]: Set `unique` same as `chart-operator`.

**Comment by tomahawk28** [comment]
I only updated template but now I got architect error during circleCI testing. 🤔

File: .circleci/config.yml [hunk 2]
```diff
@@ -67,10 +57,57 @@ workflows:
...
-          name: deploy
+      # build and push chart to app catalog
+      - architect/push-to-app-catalog:
+          name: push-app-operator-to-control-plane-app-catalog
+          app_catalog: "control-plane-catalog"
+          app_catalog_test: "control-plane-test-catalog"
+          chart: "app-operator"
+          # Make sure docker image is successfully built.
+          requires:
+            - push-app-operator-to-quay
```
Comment (rossf7) [inline .circleci/config.yml:68]: Please check against the feedback in the chart-operator PR. We should wait for Aliyun as well.

https://github.com/giantswarm/chart-operator/pull/421/files#r409757080

File: .circleci/config.yml [hunk 3]
```diff
@@ -67,10 +57,57 @@ workflows:
...
+          app_name: "app-operator"
+          app_collection_repo: "kvm-app-collection"
+          unique: "true"
+          requires:
+            - push-app-operator-to-control-plane-app-catalog
+          filters:
+            branches:
+              ignore: /.*/
+            tags:
+              only: /^v.*
```
Comment (rossf7) [inline .circleci/config.yml:113]: There are a few line endings that need to be fixed.

**Comment by rossf7** [comment]
> I only updated template but now I got architect error during circleCI testing. 🤔

> wget: missing URL

https://circleci.com/gh/giantswarm/app-operator/4046?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

I think we hit the GitHub rate limiting again.

File: .circleci/config.yml [hunk 4]
```diff
@@ -21,16 +21,6 @@ jobs:
...
-  deploy:
-    machine: true
-    steps:
-    - checkout
-    - attach_workspace:
-        at: .
-    - deploy:
-        command: |
-            ./architect deploy
-
```
Comment (rossf7) [inline .circleci/config.yml:33]: We need to keep the deploy. So we can still deploy with draughtsman.

File: helm/app-operator/templates/deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  replicas: 1
+  revisionHistoryLimit: 3
+  strategy:
+    type: Recreate
+  template:
+    metadata:
+      labels:
+        {{- include "labels.selector" . | nindent 8 }}
+      annotations:
+        releasetime: {{ $.Release.Time }}
```
Comment (rossf7) [inline helm/app-operator/templates/deployment.yaml:21]:
```suggestion
        releasetime: {{ $.Release.Revision }}
```

We need this for Helm 3.

File: helm/app-operator/templates/service-account.yaml [hunk 6]
```diff
@@ -0,0 +1,7 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ include "resource.default.name" . }}
+  namespace: {{ include "resource.default.namespace" . }}
+  labels:
+    {{- include "labels.common" . | nindent 4 }}
```
Comment (rossf7) [inline helm/app-operator/templates/service-account.yaml:7]: Line ending.

File: helm/app-operator/values.yaml [hunk 7]
```diff
@@ -0,0 +1,34 @@
+# Default values for app-operator-chart.
```
Comment (rossf7) [inline helm/app-operator/values.yaml:1]:
```suggestion
# Default values for app-operator chart.
```

File: helm/app-operator/templates/deployment.yaml [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+    spec:
+      volumes:
+      - name: {{ .Chart.Name }}-configmap
```
Comment (kopiczko) [inline helm/app-operator/templates/deployment.yaml:24]: Let's replace `{{ .Chart.Name }}` with `{{ include "name" . }}`

File: helm/app-operator/templates/deployment.yaml (hunk 5 above)
Comment (tomahawk28) [reply helm/app-operator/templates/deployment.yaml:21]: I will rename the field to `releaseRevision`

File: helm/app-operator/values.yaml (hunk 7 above)
Comment (rossf7) [reply helm/app-operator/values.yaml:1]: This needs updating because the chart was renamed.

Comment (tomahawk28) [reply helm/app-operator/values.yaml:1]: https://github.com/giantswarm/app-operator/pull/240/files#diff-540ebd3b9c05ac90abbd6b0cb458468eR1 This is there already, I don't know why github didn't show it as outdated :(.

Comment (rossf7) [reply helm/app-operator/values.yaml:1]: Strange. Thanks for confirming. 👍

**Pull request merged**